### PR TITLE
Switch to maven 3 and update dependencies and plugins

### PR DESCRIPTION
**Pull request by dacrome**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -118,113 +101,64 @@
...
             <groupId>org.apache.maven</groupId>
             <artifactId>maven-plugin-api</artifactId>
             <version>${maven.version}</version>
-            <scope>compile</scope>
         </dependency>
 
         <dependency>
             <groupId>org.apache.maven</groupId>
             <artifactId>maven-project</artifactId>
-            <version>${maven.version}</version>
```
Comment (woglinde) [inline pom.xml]: Why is the version not needed here anymore?

File: src/main/java/de/tarent/maven/plugins/pkg/upload/WagonUploader.java [hunk 2]
```diff
@@ -3,21 +3,20 @@
 import java.io.File;
 
 import org.apache.maven.execution.MavenSession;
+import org.apache.maven.plugin.BuildPluginManager;
```
Comment (woglinde) [inline src/main/java/de/tarent/maven/plugins/pkg/upload/WagonUploader.java]: The test needs to be changed to:

testConstructor(de.tarent.maven.plugins.pkg.upload.WagonUploaderTest): org.apache.maven.plugin.DefaultBuildPluginManager cannot be cast to org.apache.maven.plugin.PluginManager

File: pom.xml (hunk 1 above)
Comment (dacrome) [reply pom.xml]: ups, actually I set it to 2.0.11 but we do not need the whole dependency anymore. I will remove it.

Comment (dacrome) [reply pom.xml]: done

File: src/main/java/de/tarent/maven/plugins/pkg/upload/WagonUploader.java (hunk 2 above)
Comment (dacrome) [reply src/main/java/de/tarent/maven/plugins/pkg/upload/WagonUploader.java]: now all tests are green

**Comment by woglinde** [comment]
Thanks, for the pull request.

**Pull request merged**